scripts/stitching/stitching.py

```diff
@@ -32,203 +35,291 @@ def combine_entity_by_name(roi_fov_paths, cols, rows, enforce_square=False, marg
+        if margin is None:
+            margin = {
+                "x":int(np.round(r["xMargin"]*px_per_u)),
+                "y":int(np.round(r["yMargin"]*px_per_u))
+            }
+        elif type(margin) != dict:
+            if type(margin) != list:
+                margin = {"x":margin, "y":margin}
+            else:
+                margin = {"x":margin[0], "y":margin[1]}
+        cols, rows = [], []
+        um_min_x, um_min_y = 999999999, 999999999
+        roi_fov_paths=[]
+        for f in r["fovs"]:
+            fov_name = f["name"]
+            cols.append(f["gridPosition"]["x"]+1)
+            rows.append(f["gridPosition"]["y"]+1)
+            coord = f["centerPointMicrons"]
+            if coord["x"] < um_min_x:
+                um_min_x = coord["x"]
+            if coord["y"] < um_min_y:
+                um_min_y = coord["y"]
+            roi_fov_paths.append(
+                os.path.join(
+                    run_folder,
+                    "fov" + "-" + str(
+                        f["runOrder"]).zfill(2) + "-" + fov_name) + ".tiff")
+    
+        print(f'Stitching {run} - {roi}: {len(roi_fov_paths)} FOVs')
+        out_img, panel, max_dim, ref_metadata = combine_entity_by_name(
+            roi_fov_paths, cols, rows, enforce_square=enforce_square,
+            margin=margin)
+        out_img = out_img.astype(np.uint8, copy=False)
+
+        metadata = ref_metadata.copy()
+        metadata["coordinates"] = (um_min_x, um_min_y)
+        metadata["size"] = max_dim/px_per_u
+        metadata["fov_name"] = roi
+
+        out_mibi_tiff = mi.MibiImage(
+            out_img, panel, datetime_format='%Y-%m-%d', **metadata)
         
-            print(run, roi, f"{len(roi_fov_paths)} FOVs")
-            try:
-                out_img, panel, dims, ref_metadata = combine_entity_by_name(roi_fov_paths, cols, rows, 
-                                                                            enforce_square=enforce_square, 
-                                                                            margin=margin)
-            except:
-                continue
-            out_img = out_img.astype(np.uint8, copy=False)
-
-            metadata = ref_metadata.copy()
-            metadata["coordinates"] = (um_min_x, um_min_y)
-            metadata["fov_name"] = roi
-
-            out_mibi_tiff = mi.MibiImage(
-                out_img, panel, datetime_format='%Y-%m-%d', **metadata)
-            if dims[0] != dims[1]:  # non-square
-                out_mibi_tiff.set_image_size((dims[0]/px_per_u, dims[1]/px_per_u))
-            else:  # square
-                out_mibi_tiff.set_image_size(dims[0]/px_per_u)
-            
-            f_split = out_mibi_tiff.folder.split('/')
-            f_split[0] = roi
-            out_mibi_tiff.set_fov_id(f_split[0], '/'.join(f_split))
+        f_split = out_mibi_tiff.folder.split('/')
+        f_split[0] = roi
+        out_mibi_tiff.set_fov_id(f_split[0], '/'.join(f_split))
 
+        if not out_folder:
```

Comment:
Made `out_folder` an optional parameter, if omitted then it defaults to saving in the `run_folder`

Reply:
See comment